nemo/collections/nlp/models/nlp_model.py

```diff
@@ -397,7 +397,21 @@ def dummy():
                         model.trainer.strategy.launcher.launch(dummy, trainer=model.trainer)
                     model.trainer.strategy.setup_environment()
                 sharded_state_dict = model.sharded_state_dict()
-                checkpoint['state_dict'] = sharded_state_dict
+                if kwargs.get("load_mlm", False):
+                    mlm_sharded_state_dict = {}
+                    for k, v in sharded_state_dict.items():
+                        # Remove 'model.' from the sharded_state_dict keys
+                        new_key = k.replace('model.', '', 1)
+
+                        # Update the key attribute of the ShardedTensor value
+                        new_value = v
+                        new_value.key = v.key.replace('model.', '', 1) if hasattr(v, 'key') else v.key
```

Comment:
@mikolajblaz yes, when I'm trying to load mlm ckpt as nemo ckpt without removing that "model." suffix It gives me KeyError on that step because it's trying to find keys in the checkpoint with "model." suffix.

and the structure looks like this:
{`model.decoder.somelayername`: ShardedTensor(key='model.decoder.somelayername', tensor=tensorvalues)}

so we also want to remove it for tensor keys